### PR TITLE
AB Test: Wrapping up Checkout UX improvements

### PR DESCRIPTION
**Pull request by victorespigares**
#### Changes proposed in this Pull Request

* Removing `CreditCardFormFieldsUX` component and implementing the proposed changes to  `CreditCardFormFields` as new control.

#### Testing instructions

Visit the URL:  https://calypso.live/?branch=update/checkout-implementing-ux-improvements

1. Go to a testing site or create a new one.

2. Purchase or upgrade to a plan to trigger Checkout. Also, you can directly go to a Checkout URL such as http://calypso.live/checkout/[YOUR_SITE]/blogger as seen [here](https://github.com/Automattic/wp-calypso/tree/master/client/my-sites/checkout)

4. Confirm you're seeing the Checkout form with UX improvements. Pay attention to placeholders and gray text explanations on `"Cardholder Name"` and `"Security Code"`.

image

5. Go to /me -> Manage Purchases -> Billing History -> Add Credit Card ([link](https://calypso.live/me/purchases/billing)). Make sure the Credit Card form is the UX-improved one.

### CODE REVIEW

File: client/components/credit-card-form-fields/index.jsx [hunk 1]
```diff
@@ -145,40 +149,48 @@ export class CreditCardFormFields extends React.Component {
 			<div className="credit-card-form-fields">
 				{ this.createField( 'name', Input, {
 					autoFocus,
-					label: translate( 'Cardholder Name', {
+					label: translate( 'Cardholder Name {{nameExplainer/}}', {
```
Comment (niranjan-uma-shankar) [inline client/components/credit-card-form-fields/index.jsx]: Using `nameExplainer ` component here has two issues:
1. The component makes another translate call, and having nested `translate()` looks like something that can be avoided, just for reduced complexity and cleaner flow.
2. The second translate call is missing a translator comment, so `(as written on card)` could be hard to get the context for a translator. Of course, we could add a comment here, but refer previous point, it makes it more complex.

I would suggest rewriting this as:

```
label: translate( 'Cardholder Name {{span}}(as written on card){{/span}}', {
	comment: 'Cardholder name label on credit card form',
	components: {
		span: <span className="credit-card-form-fields__explainer" />,
	},
} ),
```

The advantage is that _(as written on card)_ is now part of the original string, so a translator gets the full context.

File: client/components/credit-card-form-fields/index.jsx [hunk 2]
```diff
@@ -145,40 +149,48 @@ export class CreditCardFormFields extends React.Component {
...
 					} ) }
 
 					{ this.createField( 'cvv', Input, {
 						inputMode: 'numeric',
-						placeholder: translate( 'CVV', {
-							comment: '3 digit security number on credit card form',
-						} ),
-						label: translate( 'CVV {{infoPopover/}}', {
+						placeholder: ' ',
+						label: translate( 'Security Code {{explainerText/}} {{infoPopover/}}', {
```
Comment (niranjan-uma-shankar) [inline client/components/credit-card-form-fields/index.jsx]: Refer this comment, I suggest modelling it the same way. Remove `{{explainerText/}} ` and replace it with `{{span}}{{/span}}`. 

`{{infoPopover/}}` can remain as is.

File: client/components/credit-card-form-fields/index.jsx [hunk 3]
```diff
@@ -92,6 +92,10 @@ export class CreditCardFormFields extends React.Component {
 		this.updateFieldValues( event.target.name, event.target.value );
 	};
 
+	addExplainer = explainerText => {
```
Comment (niranjan-uma-shankar) [inline client/components/credit-card-form-fields/index.jsx]: Refer this comment, if we model it as suggested, we no longer need this function.

File: client/components/credit-card-form-fields/index.jsx (hunk 1 above)
Comment (victorespigares) [reply client/components/credit-card-form-fields/index.jsx]: Great suggestion! Addressed in https://github.com/Automattic/wp-calypso/pull/33034/commits/25a06704d7abfa647067fcc06631d7846a6c9936.

File: client/components/credit-card-form-fields/index.jsx (hunk 2 above)
Comment (victorespigares) [reply client/components/credit-card-form-fields/index.jsx]: Addressed in https://github.com/Automattic/wp-calypso/pull/33034/commits/25a06704d7abfa647067fcc06631d7846a6c9936.

File: client/components/credit-card-form-fields/index.jsx (hunk 3 above)
Comment (victorespigares) [reply client/components/credit-card-form-fields/index.jsx]: Addressed in https://github.com/Automattic/wp-calypso/pull/33034/commits/25a06704d7abfa647067fcc06631d7846a6c9936.

**Pull request merged**